packages/codemirror/src/editor.ts

```diff
@@ -615,7 +631,17 @@ class CodeMirrorEditor implements CodeEditor.IEditor {
       return;
     }
     this._changeGuard = true;
-    this._model.value.text = this.doc.getValue();
+    let value = this._model.value;
+    let start = doc.indexFromPos(change.from);
```

Comment:
What I mean is that `doc.indexFromPos` is using the *current* buffer, whereas `.from` is from the *previous* buffer.